sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
             .addRelationshipWithResponseAsync(digitalTwinId, relationshipId, relationship, context)
-            .flatMap(
-                response -> {
-                    try {
-                        String jsonResponse = mapper.writeValueAsString(response.getValue());
-                        return Mono.just(new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse));
-                    } catch (JsonProcessingException e) {
-                        return Mono.error(e);
-                    }
-                });
+            .map(response -> {
+                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+                DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                return new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders);
+            });
+    }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @return The application/json relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<String> getRelationship(String digitalTwinId, String relationshipId) {
+        return withContext(context -> getRelationship(digitalTwinId, relationshipId, context))
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Gets a relationship on a digital twin.
+     *
+     * @param digitalTwinId The Id of the source digital twin.
+     * @param relationshipId The Id of the relationship to retrieve.
+     * @return A REST response containing the application/json relationship corresponding to the provided relationshipId.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<String>> getRelationshipWithResponse(String digitalTwinId, String relationshipId) {
+        return withContext(context -> getRelationship(digitalTwinId, relationshipId, context));
+    }
+
+    Mono<DigitalTwinsResponse<String>> getRelationship(String digitalTwinId, String relationshipId, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .getRelationshipByIdWithResponseAsync(digitalTwinId, relationshipId, context)
+            .flatMap(response -> {
+                try {
+                    String jsonResponse = mapper.writeValueAsString(response.getValue());
+                    DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                    return Mono.justOrEmpty(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, twinHeaders));
```

Comment:
why justOrEmpty?

Reply:
REturns the mono containing the item, or an empty mono, if no item is retrieved.